launch/test/launch/utilities/test_signal_management.py

```diff
@@ -25,19 +25,19 @@
 
 
 def cap_signals(*signals):
-    def _noop(*args):
-        pass
-
     def _decorator(func):
         @functools.wraps(func)
         def _wrapper(*args, **kwargs):
             handlers = {}
             try:
                 for s in signals:
-                    handlers[s] = signal.signal(s, _noop)
+                    handlers[s] = signal.signal(s, signal.default_int_handler)
                 return func(*args, **kwargs)
+            except KeyboardInterrupt:
+                pass
```

Comment:
@ciandonovan hmm, this will skip the entire test with a green check. The `asyncio` loop won't deliver any handlers if it is interrupted.

Reply:
Could you clarify why, or what it's interrupting? My understanding is that the `assert` in the `finally` section will always get run no matter what exceptions occur so long as they're handled.

Reply:
Right, those `assert` will be executed (and likely succeed), but all the `assert` in the `test_async_safe_signal_manager` body after the first signal is raised won't because a `KeyboardInterrupt` exception will be raised during `asyncio.run_until_complete`.